pymagicc/io.py

```diff
@@ -2095,3 +2119,106 @@ def get_generic_rcp_name(inname):
     except KeyError:
         error_msg = "No generic name for input: {}".format(inname)
         raise ValueError(error_msg)
+
+
+# TODO: move out of here in clean up, see issue #166
+def join_timeseries(base, overwrite, join_linear=None):
+    """Join two sets of timeseries
+
+    Warning: this function has not been tested with data which has anything other than
+    years in its time axis.
+
+    Parameters
+    ----------
+    base : :obj:`pd.DataFrame`
+        Base timeseries to use
+
+    overwrite : :obj:`pd.DataFrame`
+        Timeseries to join onto base. Any points which are in both `base` and
+        `overwrite` will be taken from `overwrite`.
+
+    join_linear : tuple of len(2)
+        A list/array which specifies the period over which the two timeseries should
+        be joined. The first element is the start time of the join period, the second
+        element is the end time of the join period. In the join period (excluding the
+        start and end times), output data will be a linear interpolation between (the
+        annually interpolated) `base` and `overwrite` data. If None, no linear join
+        will be done and any points in (the annually interpolated) `overwrite` data
+        will simply overwrite any points in `base`.
+
+    Returns
+    -------
+    :obj:`MAGICCData`
+        The joint timeseries. The resulting data is linearly interpolated onto annual steps
+    """
+    if join_linear is not None:
+        if len(join_linear) != 2:
+            raise ValueError("join_linear must have a length of 2")
+
+    result = _join_timeseries_mdata(base, overwrite, join_linear)
+
+    return result
+
+
+def _join_timeseries_mdata(base, overwrite, join_linear):
+    min_year = min(base["time"].min(), overwrite["time"].min())
+    max_year = max(base["time"].max(), overwrite["time"].max())
+    new_index = range(min_year, max_year + 1)
+
+    result = _reindex_df(base, new_index)
+    ow = _reindex_df(overwrite, new_index)
+
+    if result.stack().index.intersection(ow.stack().index).empty:
+        raise ValueError("No overlapping indices, a simple append will do")
```

Comment:
Should this raise an exception? The function is meant to join time series so why not just return the appended dataframes in this case?

Reply:
My reasoning here was that if you call join_timeseries, you expect there to be some overwriting done and should be told if that's not the case. Maybe the function needs to be renamed to make that clearer?

Reply:
Or just raise a warning instead?

Reply:
It's to guard against the example in the notebook where you think you're joining scenarios, except something stupid has gone wrong (i.e. the todo column isn't behaving) and so you end up not joining anything and just getting a massive dataframe back which you can't use.

Reply:
Ahh I see. That raises another question - what is the point of N/A in the TODO column if you just set it to set??? 

Perhaps, there needs to be a higher level function which takes scenario filenames and tries to merge them as it is a bit clunky having to load the scenarios and set the TODO column to set and then call join_timeseries

Reply:
> That raises another question - what is the point of N/A in the TODO column if you just set it to set???

I had this idea that cause they were data files, the user should have to choose what to do with them. On reflection, reading in SET makes more sense.

Will tweak and right high level option now